fe/fe-core/src/main/java/org/apache/doris/cloud/backup/CloudRestoreJob.java

```diff
@@ -0,0 +1,649 @@
+            super.run();
+        } catch (UserException e) {
+            LOG.error("failed to run cloud restore job", e);
+        }
+    }
+
+    @Override
+    public void checkIfNeedCancel() {
+        super.checkIfNeedCancel();
+        if ((cloudClusterId = ((CloudSystemInfoService) Env.getCurrentSystemInfo()).getCloudClusterIdByName(
+                cloudClusterName)) == null) {
+            status = new Status(Status.ErrCode.NOT_FOUND, "cluster " + cloudClusterName
+                    + " has been removed");
+        }
+    }
+
+    @Override
+    public void checkStorageVault(OlapTable localTable) {
+        Preconditions.checkNotNull(storageVaultName);
+        if (((CloudEnv) Env.getCurrentEnv()).getEnableStorageVault()) {
+            if (localTable.getStorageVaultId().isEmpty()) {
+                status = new Status(Status.ErrCode.COMMON_ERROR, "local table " + localTable.getName()
+                        + " has no storage vault.");
+                return;
+            }
+            String localStorageVaultName = localTable.getStorageVaultName();
+            if (!localStorageVaultName.equals(storageVaultName)) {
+                // currently we only support unique storage vault name in one restore job
+                status = new Status(Status.ErrCode.COMMON_ERROR,
+                        "local table " + localTable.getName() + " storage vault is " + localStorageVaultName
+                                + ", but restore job storage vault is " + storageVaultName);
+            }
+        }
+    }
+
+    @Override
+    public void doCreateReplicas() {
+        try {
+            handleMetaObject(MetaSeriviceOperation.PREPARE);
+            // send create tablets requests
+            boolean needSetStorageVault = ((CloudEnv) Env.getCurrentEnv()).getEnableStorageVault();
+            for (Map.Entry<OlapTable, Cloud.CreateTabletsRequest.Builder> entry : tabletsPerTable.entrySet()) {
+                OlapTable table = entry.getKey();
+                Cloud.CreateTabletsRequest.Builder requestBuilder = entry.getValue();
+                Cloud.CreateTabletsResponse resp = sendCreateTabletsRequests(requestBuilder, table,
+                        needSetStorageVault);
+                if (resp.hasStorageVaultId()) {
+                    storageVaultId = resp.getStorageVaultId();
+                    needSetStorageVault = false;
+                }
+            }
+            // set storage vault for new restoring table
+            for (Table table : restoredTbls) {
+                if (table.getType() == TableIf.TableType.OLAP) {
+                    OlapTable olapTable = (OlapTable) table;
+                    if (olapTable.getStorageVaultId().isEmpty() && storageVaultId != null) {
+                        olapTable.setStorageVaultId(storageVaultId);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            status = new Status(Status.ErrCode.COMMON_ERROR, e.getMessage());
```

Comment:
How do we handle created tablets?

Reply:
Please add a injest case to cover this branch.

Reply:
created tablets will be recycle by ms, because we inserted a recycle partition key into fdb here.